Added a best practices section to Pathpoint docs page (#18180)
newrelic/docs-website: src/content/docs/new-relic-solutions/business-observability/intro-pathpoint.mdx

## Creating Flows [#flows]

In previous sections we described the main elements within the pathpoint UI: stages, steps, and touchpoints.  In pathoint the highest level grouping of these elements is the *flow*.  A flow is usually used to model a "business journey" or "customer journey".   The stages of a flow represent higher level concepts that compose the journey.  When we think about what kinds of stages to use it helps to think about what our business process or "journey" is and how we think of it in sequency.   Often it helps to look at other flows within a similar industries.  Examples of industries or verticals are:

- Hotel or Hospitality
- Cruise Line or Airline
- Rideshare
- Consumer Packaged Goods
- Online Marketplace
- General Retail
- Quick Service Restaurant
- Mining, Construction, Oil/Gas
- Digital Streaming Media
- Online News Media
- Retail or Commercial Banking
- Insurance
- Talent Management

> Within an industry it is often useful to model more than one flow.  For example in the Insurance industry we may want to model separate flows for purchasing a policy and filing a claim respectively.  In fact due to how the organization is structured you may want a flow that separates home insurance from auto insurance or consumer insurance from commercial insurance.  However, it is recommend to start off with a simple flow of 4-5 steps and either break it down later or add to it to make it more complete.  Generally stage names are non-technical.  They may be industry specific but should be generally understood by any C-level or VP personnel in an organization.

Reviewer: ```suggestion
Within an industry, it's often useful to model more than one flow. For example, in the insurance industry, we may want to model separate flows for purchasing a policy and filing a claim respectively. In fact, due to how the organization is structured, you may want a flow that separates home insurance from auto insurance or consumer insurance from commercial insurance. However, we recommend to start off with a simple flow of 4-5 steps, and either break it down later or add to it to make it more complete. Generally stage names are non-technical. They may be industry specific but should be generally understood by any C-level or VP personnel in an organization.
```

<img
    title="Airport Flow Example"
    alt="Airport Flow Example"
    src={airportFlowExample}
/>

### Some Stage Examples

<table>
  <thead>
    <tr>
      <th style={{ width: "200px" }}>
        Industry
      </th>

      <th style={{ width: "200px" }}>
        Flow Description
      </th>

      <th>
        Stages
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Hotel (Hospitality)
      </td>

      <td>
        Guest Booking & Stay
      </td>

      <td>
        Browse/Search for Room, Book Room, Check-In & Stay, Check-out, Post-Check-out
      </td>
    </tr>
    <tr>
      <td>
        Online Marketplace
      </td>

      <td>
        Basic Purchase Journey
      </td>

      <td>
        Browse / Search for Product, Cart Managment, Purchase, Shipment & Returning, Notification & Marketing
      </td>

    </tr>
    <tr>
      <td>
        Rideshare
      </td>

      <td>
        Book and Take  Ride
      </td>

      <td>
        Book, Driver Assignment, Ride Start, Ride End, Payment, Feedback
      </td>

    </tr>
</tbody>
</table>

> ### Step Examples

Reviewer: ```suggestion
### Step examples [#examples]
```

> Steps are used to bridge the gap between the abstraction of stages and the detail of signals.  While stages may be understood by higher level stakeholders, steps may sometimes be more technical or operations specific.  The key thing is that steps do not require a link to a single specific signal.  Signals are where we connect to the underlying telemetry.

Reviewer: ```suggestion
Steps are used to bridge the gap between the abstraction of stages and the detail of signals. While stages may be understood by higher level stakeholders, steps may sometimes be more technical or operations specific. The key thing is that steps do not require a link to a single specific signal. Signals are where we connect to the underlying telemetry.
```

<table>
  <thead>
    <tr>
      <th style={{ width: "200px" }}>

>         Flow::Stage

Author: Yes it's common style for a class hierarchy.

      </th>

      <th style={{ width: "200px" }}>
        Steps
      </th>
    </tr>
  </thead>
 <tbody>
    <tr>
      <td>
        Rideshare::Book
      </td>
      <td>
        Login, Search, Set Pickup, Set Destination, Select Ride Type, Payment Option, Confirm Ride
      </td>
    </tr>
    <tr>
      <td>
        Rideshare::Payment
      </td>
      <td>
        Payment Initiation, Payment Processing, Verification, Payment Confirmation
      </td>
    </tr>
    <tr>
      <td>
        Rideshare::Feedback
      </td>
      <td>
        Initiate Feedback Request, Receive User Feedback, Aggregate Ratings
      </td>
    </tr>
</tbody>
</table>

> *The rideshare journey (stages and steps without signals)*

Author: That text if I'm looking at the right one is meant to describe the screenshot below it:

```*The rideshare journey (stages and steps without signals)*```

2. Journey critical back end transactions or SLIs that directly impact revenue or customer experience
3. Journey critical third party services (often obtained via Synthetics)
4. Overall health of the most important services in each step
5. Supporting infrastructure and platforms relevant to each step (often cloud services such as load balancers, container services, or managed databases or messaging systems).

For example for a *login* step you may consider these...

1. A Javascript error rate for the login page action
2. The overall latency for the login page action
3. SLIs related to backend transactions for authenticating
4. Backend transactions or services related to user lookup
5. Database health of databases used for storing user information
6. Load balancer health for a load balancer sitting in front of authentication service
7. Redis database health for a cache of user state information
8. etc.

The goal is provide signals which have a level of indepence and when an incident is going on you will be able to deduce from what is *red* what the problem may be.

<img
    title="Signal Selection"
    alt="Selection Selection"
    src={signalSelection}
/>

### Best Practices

> **Work from the top down**
> 
> The reason we start from the top down when designing our flow is we may not have all of the signals needed yet to properly cover our flow.  In the process of the developing the flow we will identify areas were we have coverage and areas where we do not have coverage.
> 
> It’s better to have a well thought out flow with some signal gaps than a flow that is put together hastily with a random set of signals.
> 
> **Feel free to use multiple flows if your Journey is very complex**
> 
> Trying to force all relevant signals into a single flow can lead to a congested and confusing signal.  Consult with internal stakeholders (and the NR account team) to see how best to organize flows.  For example* an insurance company way want to might separate the following flows:
> 
> - Finding and purchasing new policies
> - Filing and paying out insurance claims
> 
> **Don’t use a signal just because it’s available**
> 
> Be selective and focus on the signals that are most likely related to immediate business or user impact.   You may have 5 alert conditions related to an entity, however you may choose to only add 1 or 2. 
> 
> **Don’t be afraid to improve existing signals**
> 
> Flapping signals will create a noisy, flapping Pathpoint flow.  Consider refining alert thresholds or service level queries to make signals more actionable and meaningful.  If a signal is red, there should be real or at least probably risk to the business process.  
> 
> **Use Synthetics to capture external dependencies**
> 
> Your process may be responsible for 3 or 4 third party APIs or other SaaS systems.  Synthetics is the conventional way to capture the health of those systems. 
> 
> **Use Workloads to group entities**
> 
> Workloads are a powerful means to create aggregated entities where if one or several sub-entities have critical issues we want that overall workload to show as critical.  This can be done to capture large pools of related infrastructure (things like load balancers, RDBMs, other cloud services).    Another tip is to create workloads that group multiple service levels into a single aggregate service level.
> 
> **Use useful metadata tags on your entities**
> 
> It is good practice to put tags on your entities such as service levels, workloads, synthetics etc.   It is especially helpful to add flow, stage, and step tags on your entities.  This will allow you to use these signals more easily with Pathpoint.
> 
> **Use the Pathpoint KPIs row to capture important counts, $ amounts, etc. related to your flow**
> 
> Some organizations want to have counts for important critical events like user logins, user registrations, successful checkouts, carts abandoned as well as important dollar amounts like total checkout value, total returned product value.   You can usually find some of these in your existing telemetry and they don’t have to be associated with an entity.  They can be logs, custom metrics, etc.
> 
> Let’s say you have a stage called “Search” you may want to have KPI’s like:
> 
> - Successful Searchs
> - Slow Searches
> - Failed Searches
> 
> If you have stages for Order, Cart, or Checkout you may want to have KPIs like:
> 
> - Order Count
> - Failed Orders
> - Order Value
> - Abandoned Cart
> - Average Cart Value

Reviewer: ```suggestion
**Work from the top down**

When designing our flow, the reason we start from the top down is we may not have all of the signals needed yet to properly cover the flow. During flow development, we'll identify areas were we do and do not have coverage.

It’s better to have a well thought out flow with some signal gaps than a flow that's put together hastily with a random set of signals.

**Use multiple flows if your journey is very complex**

Trying to force all relevant signals into a single flow can lead to a congested and confusing signal. Consult with internal stakeholders (and the New Relic account team) to see how best to organize flows. For example, an insurance company might want to separate the following flows:

- Finding and purchasing new policies
- Filing and paying out insurance claims

**Don’t use a signal just because it’s available**

Be selective and focus on the signals that are most likely related to immediate business or user impact.  You may have 5 alert conditions related to an entity, however you may choose to only add 1 or 2. 

**Don’t be afraid to improve existing signals**

Flapping signals will create a noisy, flapping Pathpoint flow. Consider refining alert thresholds or service level queries to make signals more actionable and meaningful. If a signal is red, there should be real or at least probably risk to the business process. 

**Use Synthetics to capture external dependencies**

Your process may be responsible for 3 or 4 third-party APIs or other SaaS systems. Synthetics is the conventional way to capture the health of those systems.

**Use Workloads to group entities**

Workloads offer a powerful method to consolidate related entities into a unified view. By aggregating infrastructure components such as load balancers, RDBMS, and other cloud services, critical issues affecting individual sub-entities can be efficiently surfaced at the workload level. Additionally, workloads can be used to combine multiple service levels into a single, overarching service level for simplified monitoring and management.

**Use useful metadata tags on your entities**

It's good practice to put tags on your entities such as service levels, workloads, synthetics etc. It's especially helpful to add flow, stage, and step tags on your entities. This will allow you to use these signals more easily with Pathpoint.

**Use the Pathpoint KPIs row to capture important counts, monetary values, etc. related to your flow**

Utilize the Pathpoint KPIs row to monitor essential counts and monetary values relevant to your workflow. This includes critical events such as user logins, registrations, checkouts, and cart abandonments, as well as financial metrics like total checkout and returned product value. Many of these metrics can be derived from existing telemetry data and do not require association with specific entities. They can be sourced from logs, custom metrics, or other available data points.

Let’s say you have a stage called “Search”, you may want to have KPI’s like:

- Successful searches
- Slow searches
- Failed searches

If you have stages for order, cart, or checkout, you may want to have KPIs like:

- Order count
- Failed orders
- Order value
- Abandoned cart
- Average cart value
```

## Creating Flows [#flows]

> In previous sections we described the main elements within the pathpoint UI: stages, steps, and touchpoints.  In pathoint the highest level grouping of these elements is the *flow*.  A flow is usually used to model a "business journey" or "customer journey".   The stages of a flow represent higher level concepts that compose the journey.  When we think about what kinds of stages to use it helps to think about what our business process or "journey" is and how we think of it in sequency.   Often it helps to look at other flows within a similar industries.  Examples of industries or verticals are:

Reviewer: ```suggestion
In previous sections we described the main elements within the pathpoint UI: stages, steps, and touchpoints. In pathpoint the highest level grouping of these elements is the *flow*. A flow is usually used to model a "business journey" or "customer journey". The stages of a flow represent higher level concepts that compose the journey.  When we think about what kinds of stages to use it helps to think about what our business process or "journey" is and how we think of it in sequence. Often it helps to look at other flows within a similar industries.  Examples of industries or verticals are:
```

2. Journey critical back end transactions or SLIs that directly impact revenue or customer experience
3. Journey critical third party services (often obtained via Synthetics)
4. Overall health of the most important services in each step

> 5. Supporting infrastructure and platforms relevant to each step (often cloud services such as load balancers, container services, or managed databases or messaging systems).

Reviewer: ```suggestion
5. Supporting infrastructure and platforms relevant to each step (often cloud services such as load balancers, container services, managed databases, or messaging systems).
```

**Use useful metadata tags on your entities**

It is good practice to put tags on your entities such as service levels, workloads, synthetics etc.   It is especially helpful to add flow, stage, and step tags on your entities.  This will allow you to use these signals more easily with Pathpoint.

**Use the Pathpoint KPIs row to capture important counts, $ amounts, etc. related to your flow**

> Some organizations want to have counts for important critical events like user logins, user registrations, successful checkouts, carts abandoned as well as important dollar amounts like total checkout value, total returned product value.   You can usually find some of these in your existing telemetry and they don’t have to be associated with an entity.  They can be logs, custom metrics, etc.

Reviewer: ```suggestion
Some organizations want to have counts for important critical events like user logins, user registrations, successful checkouts, and carts abandoned, as well as important dollar amounts like total checkout value, total returned ,product value. You can usually find some of these in your existing telemetry, and they don’t have to be associated with an entity. They can be logs, custom metrics, etc.
```
